lib/ecto/changeset.ex

```diff
@@ -1012,7 +1012,20 @@ defmodule Ecto.Changeset do
           "please use put_#{tag}/4 instead"
   end
 
-  defp put_change(data, changes, errors, valid?, key, value, _type) do
+  defp put_change(data, changes, errors, valid?, key, value, nil) do
```

Comment:
(https://github.com/elixir-ecto/ecto/pull/2031#issuecomment-293809301)

> That's strange... As far as I'm aware we should always have types for all the possible fields, if there's no type, this should be an error

@michalmuskala I think you meant to make the comment here. We allow doing e.g. `|> put_change(:some_non_existing_field, 42)`, and so there won't be a corresponding type for this field. We discussed making `put_change` stricter with regards to fields it accepts here https://github.com/elixir-ecto/ecto/pull/1883 but it got rejected.